### PR TITLE
Fix DateRangeRule

### PR DESCRIPTION
**Pull request by windaishi**
### 1. Why is this change necessary?
The `DateRangeRule` does not evaluate date ranges correctly.  There are several problems that will be addressed here.

As an example the following configuration never "matches" to true:
image

```
* fromDate: 2021-01-01 00:00
* toDate: 2021-01-01 00:00
* useTime: false
```

### 2. What does this change do, exactly?
It fixes the `DateRangeRule` and adds several unit tests to ensure the correct behaviour.

### 3. Describe each step to reproduce the issue or behaviour.
Create a rule described as above. Use the current day as the date. As an example you can assign this rule to the price of a product. The price will never change.

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Framework/Rule/DateRangeRule.php [hunk 1]
```diff
@@ -30,6 +30,7 @@ public function __construct(
         bool $useTime = false
     ) {
         parent::__construct();
+
```
Comment (windaishi) [inline src/Core/Framework/Rule/DateRangeRule.php]:
```suggestion
```

**Comment by Dominik28111** [comment]
Hi @windaishi, could you please write your changelog more detailed so that you mention which methods in which classes you changed and why.

**Comment by windaishi** [comment]
@Dominik28111 I updated the changelog ;)

File: src/Core/Framework/Test/Rule/DateRangeRuleTest.php [hunk 2]
```diff
@@ -168,4 +160,204 @@ public function testIfRuleIsConsistent(): void
 
         static::assertNotNull($this->conditionRepository->search(new Criteria([$id]), $this->context)->get($id));
     }
+
+    /**
+     * @dataProvider matchDataProvider
+     * @group Dev
```
Comment (windaishi) [inline src/Core/Framework/Test/Rule/DateRangeRuleTest.php]:
```suggestion
```

**Comment by Dominik28111** [comment]
Thanks for your contribution. Your PR will be merged as soon as our internal pipeline is green.

**Pull request closed without merging**